src/main/java/edu/wisc/my/personalizedredirection/service/RedirectionServiceImpl.java

```diff
@@ -19,15 +18,15 @@
 public class RedirectionServiceImpl implements IRedirectionService {
 	protected final Logger logger = LoggerFactory.getLogger(getClass());
 
-	public String getUrl(HttpServletRequest request, UrlDataSource dataSource) throws PersonalizedRedirectionException {
+	public String getUrl(HttpServletRequest request, UrlDataSource dataSource) throws RuntimeException {
 
 		// Find search key in the header
 		String attributeToSearchFor = dataSource.getAttributeName();
 		String toFind = request.getHeader(attributeToSearchFor);
 
 		if (toFind == null || toFind.length() == 0) {
 			logger.error("Attribute " + attributeToSearchFor + " not found");
-			throw new PersonalizedRedirectionException("Attribute " + attributeToSearchFor + " not found.");
+			throw new RuntimeException("Attribute " + attributeToSearchFor + " not found.");
```

Comment:
It's a matter of style, but I'd consider a refactor to not use exceptions for expected business cases. Accessing the service with the desired user attribute missing isn't so very exceptional -- it's an expected business case where a user doesn't have the needed identity information. There's no error, so much as it's a branch in a conditional tree.

It's spiritually different than if the datasource failed on a JDBC query or whatever.

Definitely a matter of style and reasonable programers will differ on this.